code/datums/storage/storage_interface.dm

```diff
@@ -44,20 +44,35 @@
 
 /// Updates position of all UI elements
 /datum/storage_interface/proc/update_position(screen_start_x, screen_pixel_x, screen_start_y, screen_pixel_y, columns, rows)
-	var/screen_end_x = screen_start_x + columns - 1
-	var/screen_end_y = screen_start_y + rows - 1
+	var/start_pixel_x = screen_start_x * 32 + screen_pixel_x
+	var/start_pixel_y = screen_start_y * 32 + screen_pixel_y
+	var/end_pixel_x = start_pixel_x + (columns - 1) * 32
+	var/end_pixel_y = start_pixel_y + (rows - 1) * 32
 
-	cells.screen_loc = "[screen_start_x]:[screen_pixel_x],[screen_start_y]:[screen_pixel_y] to [screen_end_x]:[screen_pixel_x],[screen_end_y]:[screen_pixel_y]"
+	cells.screen_loc = spanning_screen_loc(start_pixel_x, start_pixel_y, end_pixel_x, end_pixel_y)
+	var/left_edge_loc = spanning_screen_loc(start_pixel_x + 32, start_pixel_y, end_pixel_x, end_pixel_y)
+	var/right_edge_loc = spanning_screen_loc(start_pixel_x, start_pixel_y, end_pixel_x + max(0, (columns - 2)) * 32, end_pixel_y)
+	corner_top_left.screen_loc = left_edge_loc
+	corner_top_right.screen_loc = right_edge_loc
+	corner_bottom_left.screen_loc = left_edge_loc
+	corner_bottom_right.screen_loc = right_edge_loc
 
-	corner_top_left.screen_loc = "[screen_start_x + 1]:[screen_pixel_x],[screen_start_y]:[screen_pixel_y] to [screen_end_x]:[screen_pixel_x],[screen_end_y]:[screen_pixel_y]"
-	corner_top_right.screen_loc = "[screen_start_x]:[screen_pixel_x],[screen_start_y]:[screen_pixel_y] to [screen_start_x + max(0, columns - 2)]:[screen_pixel_x],[screen_end_y]:[screen_pixel_y]"
-	corner_bottom_left.screen_loc = "[screen_start_x + 1]:[screen_pixel_x],[screen_start_y]:[screen_pixel_y] to [screen_end_x]:[screen_pixel_x],[screen_end_y]:[screen_pixel_y]"
-	corner_bottom_right.screen_loc = "[screen_start_x]:[screen_pixel_x],[screen_start_y]:[screen_pixel_y] to [screen_start_x + max(0, columns - 2)]:[screen_pixel_x],[screen_end_y]:[screen_pixel_y]"
-
-	rowjoin_left.screen_loc = "[screen_start_x]:[screen_pixel_x],[screen_start_y]:[screen_pixel_y + 27] to [screen_start_x]:[screen_pixel_x],[screen_start_y + max(0, rows - 2)]:[screen_pixel_y + 27]"
+	var/row_loc = spanning_screen_loc(start_pixel_x, start_pixel_y + 27, end_pixel_x, end_pixel_y + 27 + max(0, rows - 2) * 32)
+	rowjoin_left.screen_loc = row_loc
 	rowjoin_left.alpha = (rows > 1) * 255
 
-	rowjoin_right.screen_loc = "[screen_end_x]:[screen_pixel_x],[screen_start_y]:[screen_pixel_y + 27] to [screen_end_x]:[screen_pixel_x],[screen_start_y + max(0, rows - 2)]:[screen_pixel_y + 27]"
+	rowjoin_right.screen_loc = row_loc
 	rowjoin_right.alpha = (rows > 1) * 255
 
 	closer.screen_loc = "[screen_start_x + columns]:[screen_pixel_x - 5],[screen_start_y]:[screen_pixel_y]"
+
+/proc/spanning_screen_loc(start_px, start_py, end_px, end_py)
```

Comment:
do me a favor could you audodoc this (like what is a spanning screen loc)
also put it in `__HELPERS/screen_objs.dm` somewhere